src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -362,7 +362,14 @@ internal void SetBufferSizes(int readBufferSize, int writeBufferSize)
         public override void Flush()
         {
             if (_handle == null) InternalResources.FileNotOpen();
-            Interop.Termios.TermiosDiscard(_handle, Interop.Termios.Queue.AllQueues);
+
+            SpinWait sw = new SpinWait();
+            while (!_writeQueue.IsEmpty)
```

Comment:
I saw you removed the `readQueue.IsEmpty` verification. What would happen if the user is reading (the queue is not empty) and then we drain?